src/gui/CMakeLists.txt

```diff
@@ -1,15 +1,29 @@
 project(gui)
-find_package(Qt5 REQUIRED COMPONENTS Widgets Svg Qml Quick QuickControls2 Xml Network)
+if (Qt6_FOUND)
+    find_package(Qt6 COMPONENTS REQUIRED Widgets Svg Qml Quick QuickControls2 Xml Network)
+else()
+    set(REQUIRED_QT_VERSION "5.15.0")
+    find_package(Qt5 ${REQUIRED_QT_VERSION} COMPONENTS REQUIRED Widgets Svg Qml Quick QuickControls2 Xml Network)
+endif()
 
 if(QUICK_COMPILER)
-        find_package(Qt5QuickCompiler)
+    if (Qt6_FOUND)
+#        find_package(Qt6 COMPONENTS QuickCompiler)
+#        set_package_properties(Qt6QuickCompiler PROPERTIES
+#                DESCRIPTION "Compile QML at build time"
+#                TYPE REQUIRED
+#        )
```

Comment:
Are we not allowing the use of the QuickCompiler with Qt6?

Reply:
I have to check because I have failed to understand what they did with the compiler in Qt6 but there has been changes

Reply:
in the end, I am working on having qt6 qml modules defined in cmake so that we gain automatic qml compiler support and also static checking via qmllint